### ISSUE TITLE
Re-design of MOHA Client

### ISSUE THREAD

**Issue by IDPL-MJU** [jiksoo75/MOHA-Kafka#26]
According to Issues #24, #25, we need to re-design current MOHA Client to effectively support a real MTC application (drug repositioning simulation based on Autodock Vina).

A number of design issues should be carefully considered as followings:

1. The implementation should be generic or flexible enough to support various types of MTC applications. This means that targeting only vina application will cause other problems when we try to apply another applications. 
   - Most of MTC applications consist of many tasks by employing parameter sweeps (e.g. looping integers, discrete set of values, directory files)
   - For now, we can consider only **directory** type of sweeping parameters as in the case of drug repositioning application in Issue #24. For example, we are sweeping the protein file parameters in the vina execution while other parameters can be fixed.
   -  This directory type of parameter sweeping (i.e. generating multiple tasks based on the number of files in a directory) can be applied to most of protein-ligand docking simulations (e.g. virtual screening, drug repositioning)

1. We need to devise a simple and easy way of describing the input/output patterns of an MTC application
   - revise the JDL(Job Description File) to make MOHA Client understand the characteristics of an application
   - It should be able to describe executable (e.g. autodock_vina.sh, vina), input files (e.g. protein files, ligand files, coordinate file), output files (e.g. pdbqt, log files) so that the MOHA Client can generate multiple tasks based on these information and upload required files into the HDFS.
   - We can assume that users already generated multiple data bundles in a specific directory based on the Issue #25 if needed.

1. Creating a MOHA Job Queue and inserting tasks

   - Since all application specific files (executable, input files etc.) are residing on the node where the MOHA Client is running, from now on, the MOHA Client should create a MOHA Job Queue and based on the JDL, generate multiple tasks and insert them into the queue.
   - We need to think about how we can describe a task as a _Kafka message_ when the MOHA Client inserts multiple tasks into the queue. Also, the locations of input or executable files in the task description should reflect the locations of those files in the HDFS.
   - Finally, MOHA Client can upload required files into the HDFS as usual.

To summarize, we need to re-design the MOHA Client to include **enhanced JDL functionality** and **MOHA Job Queue management**.

**Comment by IDPL-MJU** [comment]
Review this re-design of MOHA Client and update/add contents if needed.

**Comment by nguyenngoccao** [comment]
The new update makes MOHA capable of running MTC applications likes Autodock-Vina, yet needed to revise to be more generic, by which to support common MTC applications
